simplex/block.go

```diff
@@ -84,11 +125,80 @@ func (d *blockDeserializer) DeserializeBlock(ctx context.Context, bytes []byte)
 		return nil, err
 	}
 
-	digest := hashing.ComputeHash256Array(bytes)
-
 	return &Block{
 		metadata: *md,
 		vmBlock:  vmblock,
-		digest:   digest,
+		digest:   computeDigest(bytes),
 	}, nil
 }
+
+// blockTracker is used to ensure that blocks are properly rejected, if competing blocks are accepted.
+type blockTracker struct {
+	lock sync.Mutex
```

Comment:
I think yacov's comment around needing to hold the context lock here was more around how every `VM` function expects the `*snow.Context.Lock` to be held when it is called. Not that we needed to add a new lock.

Reply:
So, in Simplex we have two main ways in which a VM can be invoked:

1. You receive a message and respond to it. This is mostly fetching blocks from the VM or indexing them. 
2. Asynchronously via the scheduler which verifies blocks and builds new blocks.

All (1) are synchronized using a lock, but (2) are not synchronized, as we don't want to block Simplex from being prevented from responding to messages or doing network I/O or writing to the WAL while building or verifying a block. 

However, we should take care of that in the avalanchego side. 

I think the simplest solution would be to make a VM wrapper that synchronizes all VM methods.